pkg/k3s/cluster-init.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/ash
+#
+# Copyright (c) 2023 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+K3S_VERSION=v1.26.3+k3s1
+KUBEVIRT_VERSION=v0.59.0
+LONGHORN_VERSION=v1.4.1
+CDI_VERSION=v1.56.0
+
+INSTALL_LOG=/var/log/install.log
+date >> $INSTALL_LOG
+
+logmsg() {
+   local MSG
+   local TIME
+   MSG="$*"
+   TIME=$(date +"%F %T")
+   echo "$TIME : $MSG"  >> $INSTALL_LOG
+}
+
+setup_cgroup () {
+   echo "cgroup /sys/fs/cgroup cgroup defaults 0 0" >> /etc/fstab
+}
+
+
+                       
+check_network_connection () {
+ network_available=false                                               
+ while true; do
+   case "$(curl -s --max-time 5 -I http://google.com | sed 's/^[^ ]*  *\([0-9]\).*/\1/; 1q')" in
```

Comment:
Is google really needed or just a "reliable" source to check internet connectivity? I think it's not too hard to get a network where google will eventually be blocked. If you need http then you can try at least another source to check connectivity. If you just want to test internet access, then you can try to ping some reliable servers (DNS root servers for instance, or even Google's DNS server 8.8.8.{8,4})....

Reply:
I am just checking internet access. I can change to 8.8.8.8 since to resolve google.com, we probably have to hit 8.8.8.8 first.